kolibri/core/content/api.py

```diff
@@ -231,50 +236,141 @@ class OptionalPageNumberPagination(pagination.PageNumberPagination):
     page_size_query_param = "page_size"
 
 
+def map_assessmentmetadata(obj):
+    assessmentmetadata = {}
+    assessmentmetadata["assessment_item_ids"] = obj.pop("assessment_item_ids")
+    assessmentmetadata["number_of_assessments"] = obj.pop("number_of_assessments")
+    assessmentmetadata["mastery_model"] = obj.pop("mastery_model")
+    assessmentmetadata["randomize"] = obj.pop("randomize")
+    assessmentmetadata["is_manipulable"] = obj.pop("is_manipulable")
+    if all(map(lambda x: x is None, assessmentmetadata.values())):
+        return None
+    return assessmentmetadata
+
+
+def map_files(obj):
+    keys = [
+        "id",
+        "priority",
+        "available",
+        "file_size",
+        "extension",
+        "preset",
+        "lang",
+        "supplementary",
+        "thumbnail",
+    ]
+
+    def set_urls(file):
+        download_filename = models.get_download_filename(
+            obj["title"],
+            models.PRESET_LOOKUP.get(file["preset"], _("Unknown format")),
+            file["extension"],
+        )
+        file["download_url"] = reverse(
+            "kolibri:core:downloadcontent",
+            kwargs={
+                "filename": get_content_file_name(file),
+                "new_filename": download_filename,
+            },
+        )
+        file["storage_url"] = get_local_content_storage_file_url(file)
+        return file
+
+    return list(
+        map(
+            set_urls,
+            (
+                dict(zip(keys, file_tuple))
+                for file_tuple in zip(*(obj.pop("file_" + key) for key in keys))
```

Comment:
Getting prop validation errors in KContentRenderer, since some of these values are being converted to strings. 

![image](https://user-images.githubusercontent.com/10248067/81622995-3b2f5280-93a7-11ea-932d-ff551488d00f.png)

This map in the mixin gives the expected type.

https://github.com/rtibbles/kolibri/blob/4872854d9e568b356008e2455400ffabc47ec4b4/packages/kolibri-components/src/content/mixin.js#L13-L52

Reply:
Thanks for testing - I'll come back to this, as it's not vital for string freeze.

Reply:
This should be resolved now - it was the GroupConcat being used for aggregation in SQLite that was coercing everything to a string, so just had to coerce the values back again!